src/src/com/microsoft/aad/adal/CacheKey.java

```diff
@@ -155,6 +155,14 @@ public static String createMultiResourceRefreshTokenKey(AuthenticationRequest it
         return createCacheKey(item.getAuthority(), item.getResource(), item.getClientId(), true,
                 cacheUserId);
     }
+    
+    public static String createFamilyRefreshTokenKey(final AuthenticationRequest authRequest, final String userId) {
+        if (authRequest == null) {
+            throw new IllegalArgumentException("authentication request is null");
+        }
+        
+        return createCacheKey(authRequest.getAuthority(), null, null, true, userId);
```

Comment:
hm, i thought of it. But we don't even know how server gonna implement it, are you going to return the GUID. then how do we know the GUID in advance? Are they going to change the portal to give an FamilyId just as clientId? If this is the case, that will be a breaking change for ADAL anyways, we probably need to bump the major version to support it.